### PR TITLE
Upgrade to Vue 3 and Vite

### PR DESCRIPTION
**Pull request by t11r**
Should be ready for release now.

### CODE REVIEW

**Review by paulpestov — APPROVED** [review]

**Comment by t11r** [comment]
:tada:

File: package.json [hunk 1]
```diff
@@ -18,45 +18,43 @@
...
+    "@vue/test-utils": "^2.3.2",
+    "canvas": "^2.11.2",
+    "click-outside-vue3": "^4.0.1",
+    "cypress": "^12.13.0",
+    "eslint": "^8.42.0",
+    "eslint-plugin-cypress": "^2.13.3",
+    "eslint-plugin-vue": "^9.14.1",
+    "jsdom": "^21.1.2",
+    "openseadragon": "3.0.0",
+    "sass": "^1.62.1",
```
Comment (paulpestov) [inline package.json]: Do we still need "sass" as dep?

Comment (t11r) [reply package.json]: Yes. https://vitejs.dev/guide/features.html#css-pre-processors

Comment (paulpestov) [reply package.json]: Ah ok, somehow the build still succeeded for me when I tried to remove "sass".

**Comment by paulpestov** [comment]
Another issue I found is this: 
davinci

With this error on the console:
`TypeError: Cannot read properties of null (reading 'description') ...`
Used IIIF manifest: `https://iiif.ub.uni-leipzig.de/static/collections/toplevel.json`

It seems that they use `@label` instead of `label` key. Tify should fail more gracefully here I think.

**Comment by t11r** [comment]
Good find, thanks! There are actually two bugs in the collection view, one regarding the `label`, another reading props from a manifest that is not loaded yet. Fix coming soon.

**Comment by t11r** [comment]
`@label` is invalid IIIF, see https://presentation-validator.iiif.io/validate?version=2.1&url=https://iiif.ub.uni-leipzig.de/static/collections/misc/cdvost2018.json. I would suggest to add "No label" or similar to the button, but not to show the invalid `@label`, and make sure the missing label does not break the whole app.

**Comment by paulpestov** [comment]
Maybe give it a default label like "Manifest 1", "Manifest 2" and only if something useful (label/description) exists in the response, overwrite it. This could also work for collection labels.

**Comment by t11r** [comment]
Too much code just for displaying an invalid manifest. I’m just making sure the app does not break.

**Comment by paulpestov** [comment]
looks ok to me

**Pull request merged**